pkg/kubelet/rkt/rkt.go

```diff
@@ -1108,8 +1110,29 @@ func (r *Runtime) RunInContainer(containerID string, cmd []string) ([]byte, erro
 	args := append([]string{}, "enter", fmt.Sprintf("--app=%s", id.appName), id.uuid)
 	args = append(args, cmd...)
 
-	result, err := r.runCommand(args...)
-	return []byte(strings.Join(result, "\n")), err
+	result, err := r.buildCommand(args...).CombinedOutput()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			err = (*rktExitError)(exitErr)
+		}
+	}
+	return result, err
+}
+
+// rktExitError implemets /pkg/util/exec.ExitError interface.
+type rktExitError exec.ExitError
+
+var _ utilexec.ExitError = &rktExitError{}
+
+func (r *rktExitError) Error() string {
+	return (*exec.ExitError)(r).Error()
```

Comment:
I see, that's a good reason to stick with the embeded struct. Sorry for leading you down the wrong path.